# **Dataset1 - Appliances Energy Prediction (UCI)**

**Fonte**: https://archive.ics.uci.edu/dataset/374/appliances+energy+prediction

**Situação:**

**a empresa de eficiência energética está analisando o comportamento de uma residência de baixo consumo. A equipe deseja identificar períodos de consumo elevado dos eletrodomésticos e observar quais condições de temperatura e umidade estavam presentes nesses momentos.**

1. **Carregue a amostra no Pandas e apresente head(), shape, info() e describe().**

In [ ]:
dados.head()

In [ ]:
print(f'Total de registros: {dados.shape[0]}')
print(f'Total de colunas: {dados.shape[1]}')

In [ ]:
dados.info()

In [ ]:
dados.describe()

2. **Renomeie Appliances para Consumo_Eletrodomesticos e simplifique o nome de pelo menos três atributos ambientais.**

In [ ]:
dados = dados.rename({'Appliances': 'Consumo_eletrodomesticos',
             'lights':'Luminosidade',
              'T1':'Temperatura_1',
              'RH_1': 'Umidade_1',
              'T2':'Temperatura_2',
              'RH_2':'Umidade_2',
              'T3':'Temperatura_3',
              'RH_3':'Umidade_3' }, axis=1)

print(dados.columns)

3. **Determine o maior consumo de eletrodomésticos registrado na amostra.**

In [ ]:
CMAX = dados ["Consumo_eletrodomesticos"].max()
print(f'O consumm máximo de eletrodomésticos foi de {CMAX:.2f} KW')

4. **Calcule um limiar correspondente a 70% do valor máximo e crie um DataFrame contendo somente os registros acima desse limite.**

In [ ]:
df_alto_consumo = dados[dados['Consumo_eletrodomesticos'] > C70]
print(f'O limiar correspondente a 70% é de {C70:.2f} KW')

5. **Conte os registros selecionados e calcule qual percentual da amostra eles representam.**

In [ ]:
qtd_registros = len(df_alto_consumo)
percentual = (qtd_registros / len(dados)) * 100

print(f"Registros com consumo acima de 70%: {qtd_registros}")
print(f"Percentual da amostra: {percentual:.2f}%")

6. **Calcule a temperatura média de T1 e crie outro DataFrame contendo simultaneamente consumo acima de 70% do máximo e temperatura acima da média.**

In [ ]:
temp_media_t1 = dados['Temperatura_1'].mean()
df_consumo_e_temp = dados[
    (dados['Consumo_eletrodomesticos'] > C70) &
    (dados['Temperatura_1'] > temp_media_t1)
]

print(f"Média da Temperatura_1 (T1): {temp_media_t1:.2f}°C")
print(f"Registros com consumo > 70% E temp > média: {len(df_consumo_e_temp)}")

7.1. **Compare os dois DataFrames**

In [ ]:
print("\n--- Comparação entre os DataFrames ---")
print(f"DataFrame Ex. 4 (Apenas Consumo Elevado): {len(df_alto_consumo)} registros")
print(f"DataFrame Ex. 6 (Consumo Elevado + Temp1 > Média): {len(df_consumo_e_temp)} registros")

7.2. **explique o efeito da inclusão da temperatura como segundo critério.**

**A inclusão da temperatura como segundo critério reduz o número de registros analisados porque o operador lógico exige o atendimento simultâneo de duas condições. Enquanto o primeiro filtro seleciona isoladamente todos os momentos de consumo extremo de energia, a adição da temperatura interna acima da média elimina os picos de consumo ocorridos em períodos frios ou amenos. Na prática, esse refinamento afunila a amostra para isolar exclusivamente os episódios de alto consumo energético que coincidiram com um ambiente mais quente que o habitual, permitindo à equipe avaliar se o uso intenso dos eletrodomésticos possui relação direta com o aumento da temperatura na residência**

# **Dataset 2 — Steel Industry Energy Consumption (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/851/steel+industry+energy+consumption

**Situação:**

**A equipe de gestão de energia de uma indústria siderúrgica deseja localizar situações de consumo elevado e verificar se esses períodos coincidem com condições de carga elevada ou valores menos favoráveis de fator de potência.**


1.   **Carregue a amostra e renomeie Usage_kWh para Consumo_kWh, simplificando também os nomes dos atributos de fator de potência.**  


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df2 = pd.read_csv("/content/EX02_AMOSTRA.csv")

In [ ]:
df2.columns

Index(['date', 'Usage_kWh', 'Lagging_Current_Reactive.Power_kVarh',
       'Leading_Current_Reactive_Power_kVarh', 'Lagging_Current_Power_Factor',
       'Leading_Current_Power_Factor', 'WeekStatus', 'Day_of_week',
       'Load_Type'],
      dtype='object')

In [ ]:
df2.rename({
    'Usage_kWh': 'Consumo_kWh',
    'Lagging_Current_Power_Factor': 'Fator_De_Potencia_Atrasada',
    'Leading_Current_Power_Factor': 'Fator_De_Potencia_Capacitiva'
}, axis=1, inplace=True)

In [ ]:
df2.columns

Index(['date', 'Consumo_kWh', 'Lagging_Current_Reactive.Power_kVarh',
       'Leading_Current_Reactive_Power_kVarh', 'Fator_De_Potencia_Atrasada',
       'Fator_De_Potencia_Capacitiva', 'WeekStatus', 'Day_of_week',
       'Load_Type'],
      dtype='object')

2.    **Apresente uma inspeção inicial com head(), shape, info() e describe().**


In [ ]:
df2.head()

,date,Consumo_kWh,Lagging_Current_Reactive.Power_kVarh,Leading_Current_Reactive_Power_kVarh,Fator_De_Potencia_Atrasada,Fator_De_Potencia_Capacitiva,WeekStatus,Day_of_week,Load_Type
0,29/04/2018 07:15,2.88,3.82,0.0,60.20,100.00,Weekend,Sunday,Light_Load
1,04/10/2018 12:00,60.77,48.02,0.0,78.46,100.00,Weekday,Thursday,Maximum_Load
2,26/01/2018 11:30,120.42,59.65,0.0,89.61,100.00,Weekday,Friday,Maximum_Load
3,02/06/2018 14:30,3.13,0.00,16.6,100.00,18.53,Weekend,Saturday,Light_Load
4,07/12/2018 15:00,58.86,20.99,0.0,94.19,100.00,Weekday,Friday,Medium_Load


In [ ]:
df2.shape

(3504, 9)

In [ ]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3504 entries, 0 to 3503
Data columns (total 9 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   date                                  3504 non-null   object 
 1   Consumo_kWh                           3504 non-null   float64
 2   Lagging_Current_Reactive.Power_kVarh  3504 non-null   float64
 3   Leading_Current_Reactive_Power_kVarh  3504 non-null   float64
 4   Fator_De_Potencia_Atrasada            3504 non-null   float64
 5   Fator_De_Potencia_Capacitiva          3504 non-null   float64
 6   WeekStatus                            3504 non-null   object 
 7   Day_of_week                           3504 non-null   object 
 8   Load_Type                             3504 non-null   object 
dtypes: float64(5), object(4)
memory usage: 246.5+ KB


In [ ]:
df2.describe()

,Consumo_kWh,Lagging_Current_Reactive.Power_kVarh,Leading_Current_Reactive_Power_kVarh,Fator_De_Potencia_Atrasada,Fator_De_Potencia_Capacitiva
count,3504.000000,3504.000000,3504.000000,3504.000000,3504.000000
mean,27.246701,13.087491,3.802309,80.254241,84.463142
std,33.352371,16.399117,7.383096,19.084149,30.387569
min,2.450000,0.000000,0.000000,39.100000,12.790000
25%,3.200000,2.590000,0.000000,62.797500,99.800000
50%,4.555000,5.040000,0.000000,87.675000,100.000000
75%,52.030000,22.280000,1.690000,98.480000,100.000000
max,153.140000,87.700000,27.650000,100.000000,100.000000


3.  **Determine o maior consumo registrado e calcule um limiar de 75% do consumo máximo.**

In [ ]:
maior_consumo = df2['Consumo_kWh'].max()
print(f"O maior consumo registrado é de {maior_consumo:.2f} kWh.")

O maior consumo registrado é de 153.14 kWh.


In [ ]:
limiar_de_consumo = maior_consumo * 0.75
print(f"O limiar de 75% do consumo máximo é de {limiar_de_consumo:.2f} kWh.")

O limiar de 75% do consumo máximo é de 114.85 kWh.


4. **Crie um DataFrame contendo os registros acima desse limiar e calcule a quantidade e o percentual de registros.**

In [ ]:
df_limiar_consumo = df2[df2['Consumo_kWh'] > limiar_de_consumo]

In [ ]:
print(f"Quantidade de registros acima do limiar: {len(df_limiar_consumo)}")

Quantidade de registros acima do limiar: 65


In [ ]:
print(f"Percentual de registros acima do limiar: {(len(df_limiar_consumo) / df2.shape[0] * 100):.2f}%")

Percentual de registros acima do limiar: 1.86%


5. **Verifique quantos registros desse DataFrame pertencem à categoria Maximum Load.**

In [ ]:
qt_max_load = df_limiar_consumo[df_limiar_consumo['Load_Type'] == 'Maximum_Load']
print(f"Quantidade de registros da categoria Maximum Load no consumo elevado: {len(qt_max_load)}")

Quantidade de registros da categoria Maximum Load no consumo elevado: 32


6. **Escolha um dos atributos de fator de potência, observe seus valores e defina um limite coerente para representar valores mais baixos.**

In [ ]:
df2['Fator_De_Potencia_Atrasada'].describe()

,Fator_De_Potencia_Atrasada
count,3504.000000
mean,80.254241
std,19.084149
min,39.100000
25%,62.797500
50%,87.675000
75%,98.480000
max,100.000000


In [ ]:
limite_fator_potencia_atrasada = 92

7. **Crie um novo DataFrame contendo simultaneamente consumo elevado e fator de potência abaixo do limite escolhido.**

In [ ]:
df_alto_consumo_fp_baixo = df_limiar_consumo[df_limiar_consumo['Fator_De_Potencia_Atrasada'] < limite_fator_potencia_atrasada]

In [ ]:
df_alto_consumo_fp_baixo.shape

(40, 9)

8. **Explique por que esse segundo conjunto de registros pode merecer maior atenção da equipe de energia.**

O DataFrame `df_alto_consumo_fp_baixo` merece maior atenção da equipe de energia porque ele identifica os períodos em que a indústria está enfrentando **alto consumo de energia (Consumo_kWh)** e, simultaneamente, um **fator de potência atrasada abaixo do limite estabelecido**. Essa combinação é crítica por várias razões:

1.  **Custos Elevados**: Alto consumo de energia naturalmente implica em custos operacionais mais altos. Quando isso ocorre junto com um fator de potência baixo, os custos podem ser ainda maiores devido a:
    *   **Perdas na Rede**: Um fator de potência baixo significa que uma corrente elétrica maior é necessária para fornecer a mesma quantidade de potência ativa (útil). Isso causa mais perdas por aquecimento nos condutores e transformadores.
    *   **Multas ou Penalidades**: Muitas concessionárias de energia aplicam multas a consumidores industriais que operam com fator de potência abaixo de um determinado limite (geralmente 0,92 ou 0,95), pois isso sobrecarrega a rede de distribuição.

2.  **Ineficiência Operacional**: Um fator de potência baixo indica que a energia reativa é excessiva em relação à energia ativa, o que representa um uso ineficiente da eletricidade. A equipe de energia pode investigar a causa desse baixo fator de potência (por exemplo, excesso de motores indutivos sem compensação) e implementar soluções.

3.  **Potencial de Otimização**: A identificação desses períodos específicos permite que a equipe concentre seus esforços de otimização. Ao invés de tentar melhorar a eficiência em toda a operação, eles podem focar nos horários ou condições que mais contribuem para o desperdício de energia e custos adicionais. Isso pode incluir a instalação de bancos de capacitores para correção do fator de potência ou ajustes no agendamento de cargas.

Em resumo, esses registros apontam para as situações mais problemáticas em termos de eficiência energética e custos, tornando-os prioritários para análise e intervenção da equipe de gestão de energia.

# **Dataset 3 — Power Consumption of Tetouan City (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/849/power+consumption+of+tetouan+city

**Situação:**

**Uma equipe responsável pelo planejamento da distribuição de energia em Tétouan deseja identificar qual das três zonas apresenta o maior pico de consumo e observar as condições ambientais registradas nos momentos de maior demanda.**

1. **Renomeie as três variáveis de consumo para Consumo_Zona_1, Consumo_Zona_2 e Consumo_Zona_3.**

In [ ]:
df3 = pd.read_csv("/content/Tetuan_City_power_consumption_sample.csv")

In [ ]:
df3 = df3.rename({'Zone 1 Power Consumption': "Consumo_Zona_1",
                  'Zone 2  Power Consumption': "Consumo_Zona_2",
                  'Zone 3  Power Consumption': "Consumo_Zona_3"}, axis = 1)

In [ ]:
df_consumo = df3[['Consumo_Zona_1','Consumo_Zona_2','Consumo_Zona_3']]

In [ ]:
df_consumo.columns

Index(['Consumo_Zona_1', 'Consumo_Zona_2', 'Consumo_Zona_3'], dtype='object')

2. **Determine o consumo máximo registrado em cada uma das três zonas.**

In [ ]:
valor_consumo_maximo = df_consumo.stack().max()
indice_consumo_maximo = df_consumo.stack().idxmax()

# O índice retornado por idxmax será uma tupla: (indice_linha, nome_coluna)
linha_consumo_maximo = indice_consumo_maximo[0]
zona_consumo_maximo = indice_consumo_maximo[1]

3. **Identifique qual zona apresenta o maior pico de consumo na amostra.**

In [ ]:
# Extrai o número da zona a partir do nome da coluna
numero_zona = zona_consumo_maximo.split('_')[-1]

print(f"O pico de consumo (valor: {valor_consumo_maximo:.2f}) pertence à zona: {numero_zona}")

O pico de consumo (valor: 51571.50) pertence à zona: 1


4. **Para a zona identificada, calcule 70% do valor máximo e crie um DataFrame com os registros acima desse limiar.**

In [ ]:
limiar_70 = valor_consumo_maximo * 0.70

In [ ]:
# Cria o novo DataFrame filtrando apenas os registros da zona do pico
df_filtrado = df3[df3[zona_consumo_maximo] > limiar_70]

# Exibe as informações e os primeiros registros do novo DataFrame
print(f"Limiar de 70% do pico ({valor_consumo_maximo:.2f}): {limiar_70:.2f}")
print(f"Total de registros encontrados acima do limiar: {len(df_filtrado)}")
print(df_filtrado[[zona_consumo_maximo]])

Limiar de 70% do pico (51571.50): 36100.05
Total de registros encontrados acima do limiar: 2388
      Consumo_Zona_1
3        42476.93617
8        37286.69202
12       40993.90728
14       37475.16093
19       40676.02649
...              ...
7848     46863.71681
7851     36898.03063
7855     48036.10619
7858     37031.18644
7860     43328.13559

[2388 rows x 1 columns]


5. **Conte os registros e calcule o percentual que eles representam.**

In [ ]:
# Contagem de registros
total_registros = len(df2)
registros_filtrados = len(df_filtrado)

# Cálculo do percentual
percentual = (registros_filtrados / total_registros) * 100

# Exibição dos resultados
print(f"Total de registros na base: {total_registros}")
print(f"Registros acima do limiar: {registros_filtrados}")
print(f"Percentual representado: {percentual:.2f}%")

Total de registros na base: 3504
Registros acima do limiar: 2388
Percentual representado: 68.15%


6. **Calcule a temperatura média e crie um segundo DataFrame contendo consumo acima do limiar e temperatura acima da média.**

In [ ]:
# Calcula a temperatura média da base de dados
temp_media = df3['Temperature'].mean()

# Cria o segundo DataFrame aplicando ambas as condições com o operador lógico & (E)
df_filtrado_duplo = df3[(df3[zona_consumo_maximo] > limiar_70) & (df3['Temperature'] > temp_media)]

# Exibe a média e os resultados obtidos
print(f"Temperatura média: {temp_media:.2f}°C")
print(f"Registros encontrados com ambos os critérios: {len(df_filtrado_duplo)}")
print('')
print(df_filtrado_duplo[[zona_consumo_maximo, 'Temperature']])

Temperatura média: 18.79°C
Registros encontrados com ambos os critérios: 1597

      Consumo_Zona_1  Temperature
12       40993.90728        21.03
14       37475.16093        27.06
19       40676.02649        19.20
21       40256.07103        27.11
25       36511.89369        26.92
...              ...          ...
7842     42316.54485        25.11
7844     41417.14286        31.08
7848     46863.71681        24.88
7851     36898.03063        25.58
7855     48036.10619        22.31

[1597 rows x 2 columns]


7. **Compare os dois conjuntos e explique o que ocorreu com a quantidade de registros após a inclusão da condição ambiental.**

Com a inclusão da condição ambiental (Temperatura > temperatura média), a quantidade de registros diminuiu de `{{registros_filtrados_sem_condicao_ambiental}}` para `{{registros_filtrados_com_condicao_ambiental}}`.

Isso ocorreu porque a segunda condição (temperatura acima da média) agiu como um filtro adicional, removendo os registros que, embora tivessem um consumo elevado, não atendiam ao critério de temperatura. Em outras palavras, o segundo filtro restringiu ainda mais o conjunto de dados, focando apenas nos momentos de alto consumo que também estavam associados a temperaturas mais elevadas.

# **Dataset 5 — Wind & Solar Energy Production (Kaggle)**

**Fonte:** https://www.kaggle.com/datasets/ahmeduzaki/wind-and-solar-energy-production-dataset

**Situação:**

**Um operador de um portfólio de fontes renováveis deseja comparar a ocorrência de períodos de alta produção solar e alta produção eólica. Como as duas fontes possuem escalas próprias, cada uma deverá ser comparada com o seu próprio valor máximo.**

1. **Renomeie as variáveis principais para Geracao_Solar e Geracao_Eolica**

In [ ]:
df5 = pd.read_csv('/content/Energy Production Dataset.csv')

In [ ]:
# Filtragem e renomeação das variáveis/DataFrames principais conforme solicitado
Geracao_Solar = df5[df5['Source'] == 'Solar'].copy()
Geracao_Eolica = df5[df5['Source'] == 'Wind'].copy()

print(f"Registros de Geração Solar: {len(Geracao_Solar)}")
print(f"Registros de Geração Eólica: {len(Geracao_Eolica)}")

Registros de Geração Solar: 9378
Registros de Geração Eólica: 42484


2. **Determine o valor máximo registrado para cada fonte**

In [ ]:
solar_max = Geracao_Solar['Production'].max()
eolica_max = Geracao_Eolica['Production'].max()

print(f'O valor máximo registrado para a geração solar é: {solar_max:.2f} MWh')
print(f'O valor máximo registrado para a geração eólica é: {eolica_max:.2f} MWh')

O valor máximo registrado para a geração solar é: 16578.00 MWh
O valor máximo registrado para a geração eólica é: 23446.00 MWh


3. **Calcule separadamente 70% do máximo solar e 70% do máximo eólico**

In [ ]:
solar_max_70 = solar_max * 0.70
eolica_max_70 = eolica_max * 0.70

print(f'70% do valor máximo solar é: {solar_max_70:.2f} MWh')
print(f'70% do valor máximo eólico é: {eolica_max_70:.2f} MWh')

70% do valor máximo solar é: 11604.60 MWh
70% do valor máximo eólico é: 16412.20 MWh


4. **Crie um DataFrame para registros de alta geração solar e outro para registros de alta geração eólica**

In [ ]:
df_alta_geracao_solar = Geracao_Solar[Geracao_Solar['Production'] > solar_max_70]
df_alta_geracao_eolica = Geracao_Eolica[Geracao_Eolica['Production'] > eolica_max_70]

print("DataFrames de alta geração criados com sucesso.")

DataFrames de alta geração criados com sucesso.


5. **Conte os registros de cada DataFrame e calcule seus respectivos percentuais**

In [ ]:
# Contagem de registros
qtd_alta_solar = len(df_alta_geracao_solar)
qtd_total_solar = len(Geracao_Solar)

qtd_alta_eolica = len(df_alta_geracao_eolica)
qtd_total_eolica = len(Geracao_Eolica)

# Cálculo dos percentuais relativas ao total de cada fonte
pct_alta_solar = (qtd_alta_solar / qtd_total_solar) * 100
pct_alta_eolica = (qtd_alta_eolica / qtd_total_eolica) * 100

print(f"Geração Solar Alta (> {solar_max_70:.2f} MWh): {qtd_alta_solar} registros ({pct_alta_solar:.2f}% do total de registros solares)")
print(f"Geração Eólica Alta (> {eolica_max_70:.2f} MWh): {qtd_alta_eolica} registros ({pct_alta_eolica:.2f}% do total de registros eólicos)")

Geração Solar Alta (> 11604.60 MWh): 167 registros (1.78% do total de registros solares)
Geração Eólica Alta (> 16412.20 MWh): 1018 registros (2.40% do total de registros eólicos)


6. **Compare qual fonte aparece com maior frequência acima de 70% do seu próprio máximo**

In [ ]:
if pct_alta_solar > pct_alta_eolica:
    print(f"A fonte SOLAR aparece com maior frequência relativa ({pct_alta_solar:.2f}% vs {pct_alta_eolica:.2f}%).")
elif pct_alta_eolica > pct_alta_solar:
    print(f"A fonte EÓLICA aparece com maior frequência relativa ({pct_alta_eolica:.2f}% vs {pct_alta_solar:.2f}%).")
else:
    print(f"Ambas as fontes apresentam a mesma frequência relativa ({pct_alta_solar:.2f}%).")

A fonte EÓLICA aparece com maior frequência relativa (2.40% vs 1.78%).


7. **Explique por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável.**

### **Explicação Técnica**

**Por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável?**

Não é adequado aplicar um limite absoluto fixo (ex: um valor arbitrário em MW ou MWh) para ambas as fontes porque a energia solar e a energia eólica possuem **perfis de geração, fatores de capacidade e capacidades instaladas totalmente distintos**:

1. **Intermitência e Ciclo Diurno:** A energia solar opera dentro de uma janela restrita (horas de sol), atingindo picos concentrados perto do meio-dia e caindo a zero durante a noite. A energia eólica, por sua vez, pode gerar continuamente ao longo das 24 horas, variando de acordo com regimes de ventos regionais.
2. **Diferenças de Escala e Capacidade:** Se uma usina ou parque eólico tiver uma capacidade máxima pico instalada significativamente maior (ou menor) do que o parque solar, utilizar um único corte numérico distorceria a análise. Um valor alto demais poderia fazer parecer que a fonte de menor escala nunca atinge picos de alta eficiência, enquanto um valor baixo demais classificaria quase toda a produção da fonte de maior escala como 'alta'.

Ao normalizar pelo limite relativo de 70% da capacidade máxima individual de cada fonte, avalia-se o comportamento operacional e o desempenho relativo de cada tecnologia em relação ao seu próprio potencial.

# **Dataset 6 — Individual Household Electric Power Consumption (UCI)**

**Fonte:** https://archive.ics.uci.edu/dataset/235/individual+household+electric+power+consumption

**Situação:**

**Uma residência possui monitoramento elétrico detalhado e deseja identificar episódios de demanda elevada que também apresentem corrente acima do comportamento médio. O dataset é o mesmo utilizado como referência em aula, mas o critério de análise será diferente.**

1. **Carregue a nova amostra e organize os atributos utilizando nomes mais simples.**

In [ ]:
df6 = pd.read_csv('/content/EX06_AMOSTRA 1.csv')

In [ ]:
df6 = df6.rename({
                      'Global_active_power' : 'Potencia_Ativa',
                      'Global_reactive_power' : 'Potencia_Reativa',
                      'Voltage' : 'Tensao',
                      'Global_intensity' : 'Corrente',
                      'Sub_metering_1' : 'Consumo_1',
                      'Sub_metering_2': 'Consumo_2',
                      'Sub_metering_3': 'Consumo_3'}, axis=1)

2. **Determine o valor máximo de potência ativa.**

In [ ]:
PMAX = df6['Potencia_Ativa'].max()
print(f'O valor máximo de demanda energética é: {PMAX:.2f} kW')

O valor máximo de demanda energética é: 10.54 kW


3. **Calcule 75% do valor máximo e crie um DataFrame contendo registros acima desse limite.**

In [ ]:
P75 = PMAX * 0.75
df_acima_PMAX = df6[df6['Potencia_Ativa'] > P75]

4. **Calcule a quantidade e o percentual de registros selecionados.**

In [ ]:
print(f"Existem cerca de {df_acima_PMAX.shape[0]} de consumidores acima de P75")
percentual = df_acima_PMAX.shape[0] / df6.shape[0] * 100
print(f"O percentual de consumidores acima de P75 é: {percentual:.3f}%")

Existem cerca de 58 de consumidores acima de P75
O percentual de consumidores acima de P75 é: 0.028%


5. **Calcule a corrente média da amostra.**

In [ ]:
corrente_media = df6['Corrente'].mean()
print(f'A corrente média da amostra é: {corrente_media:.2f} A')

A corrente média da amostra é: 4.63 A


6. **Crie um segundo DataFrame contendo simultaneamente potência ativa acima de 75% do máximo e corrente acima da média.**

In [ ]:
df_P_Ativa_Acima = df6[(df6['Potencia_Ativa'] > P75) & (df6['Corrente'] > corrente_media)]

7. **Compare os dois conjuntos e explique o efeito da inclusão da corrente como segunda condição.**

In [ ]:
print(f"Existem cerca de {df3.shape[0]} registros com Potencia Ativa acima de P75 E Corrente acima da média.")
percentual_df3 = df3.shape[0] / df6.shape[0] * 100
print(f"O percentual desses registros é: {percentual_df3:.3f}%")

Existem cerca de 7863 registros com Potencia Ativa acima de P75 E Corrente acima da média.
O percentual desses registros é: 3.837%


A inclusão da condição de `Corrente` acima da média (`corrente_media`) **não resultou em redução no número de registros selecionados** neste caso específico.

Originalmente, com apenas a condição de `Potencia_Ativa` > `P75`, tínhamos **58 registros**, representando **0.028%** da amostra total.

Com a adição da condição `Corrente` > `corrente_media`, o número de registros permaneceu em **58**, representando **0.028%** da amostra.

Isso demonstra que, para este conjunto de dados e os limites estabelecidos, **todos os episódios de demanda elevada (Potência Ativa > P75) também apresentam uma corrente acima do comportamento médio**, o que significa que a segunda condição não filtrou registros adicionais.